Implementations/CloudFabric.Projections.Postgresql/PostgresqlProjectionRepository.cs

```diff
@@ -370,46 +375,56 @@ public async Task Upsert(Dictionary<string, object?> document, string partitionK
         
         var properties = _projectionDocumentSchema.Properties;
 
+        var queryChunk = ConstructConditionFilters(projectionQuery.Filters);
+        var fromStatements = new List<string>()
+        {
+            "\"" + TableName + "\""
+        };
+        
+        fromStatements.AddRange(queryChunk.AdditionalFromSelects);
+        
         var sb = new StringBuilder();
         sb.Append("SELECT ");
         sb.AppendJoin(',', properties.Select(p => p.PropertyName));
         sb.Append(" FROM ");
-        sb.Append("\"" + TableName + "\"");
+        sb.Append(string.Join(", ", fromStatements));
         sb.Append(" WHERE ");
 
-        var (whereClause, parameters) = ConstructConditionFilters(projectionQuery.Filters);
-
         if (!string.IsNullOrEmpty(partitionKey))
         {
-            whereClause += $" AND {nameof(ProjectionDocument.PartitionKey)} = @partitionKey";
-            parameters.Add(new("partitionKey", partitionKey));
+            queryChunk.WhereChunk += $" AND {nameof(ProjectionDocument.PartitionKey)} = @partitionKey";
```

Comment:
Should we check whether WhereChunk is empty here (in order to add or remove "AND")?